docs/zkapps/how-to-write-a-zkapp.mdx

```diff
@@ -41,88 +41,147 @@ Example projects do not create an accompanying UI.
+    ```text
+    ? Choose an example …
+  ❯ sudoku
+    tictactoe
+    ```
 
-  The created project includes the example files (the smart contract) in the project's `src/` directory. 
-  
-  To see the files that were created, change to the `sudoku` directory and run the `ls` command or open the directory in a code editor, such as VS Code.
+    Select the `sudoku` example project. 
 
-1. Run tests and see the tests pass: `npm run test`  
-  
-  To rerun tests automatically when you save changes to your code, run the tests in watch mode with `npm run testw`.
+    The created project includes the example files (the smart contract) in the project's `src` directory. 
+    
+1. View the files that were created:
 
-1. Build the example: `npm run build`
+    - Change to the `sudoku` directory.
+    - Run the `ls` command
+  
+    Or open the directory in a code editor, such as VS Code.
 
-  Compile your TypeScript into JavaScript in the project `/build` directory.
+1. Because this example zkApp was created using the zkApp CLI, tests are already included in `sudoku.test.ts`. To run tests and see the tests pass: 
 
-1. Configure your zkApp: `zk config`
+    ```sh
+    npm run test
+    ```
+    
+    To rerun tests automatically after you save changes to your code, you can run the tests in watch mode:
 
-  The command prompts guide you to add a deploy alias to your project `config.json` file. 
-  
-  The deploy alias can be anything you want. For more details, see [Deploy alias](/zkapps/tutorials/deploying-to-a-network#deploy-alias) in Tutorial 3: Deploy to a Live Network. 
-  
-  For this example on Berkeley Testnet, use:
+    ```sh
+    npm run testw
+    ```
  
-    - Deploy alias name: `berkeley` 
+1. Now that you have confirmed that tests run correctly, you can compile your TypeScript into JavaScript in the project `/build` directory. 
+
+    To build the example:
+
+    ```sh
+    npm run build
+    ```
     
-      This example uses `berkeley`, but the deploy alias name can be anything and does not have to match the network name.
-  
-    - Mina GraphQL API URL: `https://proxy.berkeley.minaexplorer.com/graphql`
+    - The `npm run build` command builds the TypeScript files in `sudoku/src` that contain the code for the smart contract.
+    - This build command compiles the TypeScript code into JavaScript in the `sudoku/build` directory.  
+
+1. Configure your zkApp:
+
+    ```sh
+    zk config
+    ```
+
+    The command prompts guide you to add a deploy alias to your project `config.json` file. 
+    
+1. Define a name for the deploy alias. 
+
+    For this example, use:
   
-    - Transaction fee to use when deploying: `0.1`
+    ```text
+    berkeley
+    ``` 
+      
+    The deploy alias name does not have to match the network name.
+
+1. Choose the target network:
+
+    ```text
+    Testnet
+    ``` 
+ 
+ 1. Set the Mina GraphQL API URL to deploy to: 
+
+    ```text
+    https://proxy.berkeley.minaexplorer.com/graphql
```

Comment:
so we want `minaexplorer` here? 
I don't fully understand what the API URL is here

Reply:
@barriebyron I think so! I believe there's only two main GraphQL endpoints that we could use for broadcasting: MinaExplorer (https://proxy.berkeley.minaexplorer.com/graphql) or MinaScan (https://berkeley.minascan.io/graphql)